src/systemd_ctypes/librarywrapper.py

```diff
@@ -15,98 +15,176 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+from __future__ import annotations
+
 import ctypes
+import inspect
+import logging
 import os
+import types
+import typing
+from typing import (
+    Annotated,
+    Any,
+    Callable,
+    Generic,
+    Optional,
+    NewType,
+    NoReturn,
+    Self,
+    Tuple,
+    TypeVar,
+)
+
+logger = logging.getLogger(__name__)
+
+if typing.TYPE_CHECKING:
```

Comment:
Note to self: https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING